### DISCUSSION TITLE
Problem of layout with validation in input or number objects

### DISCUSSION THREAD

**Discussion by huche6** [zauberzeug/nicegui#2738] [Q&A] [answered]
### Question

Hello, I am having some issues in a page layout with input objects. 

I wanted to use a validation method in an input or number objects. However, when the message appears, it modify the layout, moves the corresponding input or number field. 

As a minimal example, we can consider the example from https://nicegui.io/documentation/input
```py
from nicegui import ui

ui.input('Name', validation=lambda value: 'Too short' if len(value) < 5 else None)
ui.input('Name', validation={'Too short': lambda value: len(value) >= 5})

ui.run()
```
In this example, we can observe that the cases moves when the validation message pop-up. Couldn't it be consider as an "invisible" object even when it is not triggered? It can be problematic in a layout with multiple fields which are not aligned anymore when a validation is triggered. 
It is obviously an open question. I decided to use min and max value in number instead at the moment.

**Comment by falkoschindler** [comment, marked as answer]
Hi @huche6,

I think this problem has been discussed in #2484 and solved in PR #2492. But because the fix breaks existing layouts, we're waiting for the next major release.

**Reply by huche6** [reply]
ho ok thank you for the quick answer. I did not notice this PR sorry !
